install/kubernetes/helm/istio/charts/mixer/values.yaml

```diff
@@ -31,6 +31,12 @@ telemetry:
     # based on measurements 100ms p50 translates to p99 of under 1s. This is ok for telemetry which is inherently async.
     latencyThreshold: 100ms
 
+  gcpCredential:
```

Comment:
this looks reasonable to me.

@costinm is there a standard way of supporting multiple environments in deployment via install options?  should we be using a structure like:

```
credentials:
  gcp:
    enabled: false
    ...
```

Or is this OK until we want to support other volume mount creds?

Reply:
I don't see anything about GCP AFAIK except the name - how about a more generic way for Mixer to load a secret ? We have already too many options and I don't see any good reason to make this arbitrarily customizable.

Pick a name - mixer-secrets or something - and mount it with 'optional=true'. It can include any credentials or secrets any plugin or code may need - including the gcp one.

Since optional=true means if not present, it'll not be mounted - we don't even need
options for this.

Reply:
Rename everything into adapter-secret. I initially made it gcpCredential because mixer could have multiple adapters to different vendor backends. But I guess it is ok to have a generalized name to make adapter secret mount handy.

Reply:
Single Secret is good because it gets updated without having to restart mixer. We had this problem with ingress...